src/PackagePrivate/TypeIdsAcquirer.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Wikibase\TermStore\MediaWiki\PackagePrivate;
+
+/**
+ * Responsible for acquire unique constant ids for stored types
+ */
+interface TypeIdsAcquirer {
+
+	/**
+	 * Acquires ids of stored types, persisting the ones that do not exist yet.
+	 *
+	 * @param array $types list of types to acquire ids for
+	 *
+	 * @return array keys in returned array are the types passed in in $types
```

Comment:
int[]

Reply:
changed example here .. the interface need not specify the type of ids since it is implementation detail

Reply:
We use them as array keys, so they already can’t be anything other than strings or ints… what’s the point in making this generic?